platform/mbed_error.c

```diff
@@ -44,10 +47,46 @@ static core_util_atomic_flag error_in_progress = CORE_UTIL_ATOMIC_FLAG_INIT;
 static core_util_atomic_flag halt_in_progress = CORE_UTIL_ATOMIC_FLAG_INIT;
 static int error_count = 0;
 static mbed_error_ctx first_error_ctx = {0};
+
+#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
+    //Global for populating the context in exception handler
+    static mbed_error_ctx *const report_error_ctx=(mbed_error_ctx *)(ERROR_CONTEXT_LOCATION);
+    static bool is_reboot_error_valid = false;
+#endif
+
 static mbed_error_ctx last_error_ctx = {0};
 static mbed_error_hook_t error_hook = NULL;
 static mbed_error_status_t handle_error(mbed_error_status_t error_status, unsigned int error_value, const char *filename, int line_number, void *caller);
 
+//Helper function to calculate CRC
+//NOTE: It would have been better to use MbedCRC implementation. But
+//MbedCRC uses table based calculation and we dont want to keep that table memory
+//used up for this purpose. Also we cannot force bitwise calculation in MbedCRC 
+//and it also requires a new wrapper to be called from C implementation. Since
+//we dont have many uses cases to create a C wrapper for MbedCRC and the data
+//we calculate CRC on in this context is very less we will use a local 
+//implementation here.
+static unsigned int compute_crc32(void *data, int datalen)
+{
+    const unsigned int polynomial = 0x04C11DB7; /* divisor is 32bit */
+    unsigned int crc = 0; /* CRC value is 32bit */
+
+    for ( ;datalen>=0; datalen-- ) {
+        unsigned char b = (*(unsigned char *)data);
```

Comment:
You're not advancing the data pointer! Neater to just do

    const unsigned char *ptr = data;

at the top and use `*ptr++` in the loop.

Reply:
Good catch, fixed.